actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -537,7 +537,10 @@ def permitted_scalar_filter(params, key)
           params[key] = self[key]
         end
 
-        keys.grep(/\A#{Regexp.escape(key)}\(\d+[if]?\)\z/) do |k|
+        @@permitted_scalar_filter_regexps ||= {}
+        @@permitted_scalar_filter_regexps[key] ||= /\A#{Regexp.escape(key)}\(\d+[if]?\)\z/
```

Comment:
Won't storing it in a class variable make `Parameters` unsafe for multiple threads?